ui/src/components/Treasury.jsx

```diff
@@ -120,10 +120,10 @@ function VaultList() {
   );
 
   const vaultsList = Object.entries(vaults ?? {});
-  const vaultsToRender = vaultsList.filter(
-    entry =>
-      entry[1].status !== 'Declined' &&
-      !(!showClosed && ['Closed', 'Error in offer'].includes(entry[1].status)),
+  const vaultsToRender = vaultsList.filter(entry =>
+    showClosed
+      ? ['Closed', 'Error in offer'].includes(entry[1].status)
```

Comment:
this filter expression seems worth pulling out into a function.

how many different status values are there? could they be enumerated? if so the filter function could be O(1) by looking up in a table. Not perf critical or a blocker but something to consider.

Either way these strings seem easy to mistype so I suggest either a function or constants.

Reply:
+1 on constants, just made that change here and throughout the dapp.

I think the filter function is pretty self-documenting and only used once, it's literally just figuring out which vaults should be rendered. So I'm not too eager to make it its own function, but open to change that.

I'm not sure exactly what you mean by the O(1) table lookup, would you mind spelling out an example? We need to preprocess this list to know its length for the rendering logic below, which would be O(n). Unless you mean the `Array.includes` thing should be `Set.has` or something, but I think `Array.includes` is actually pretty fast on two elements:
![jsbench](https://user-images.githubusercontent.com/8848650/161180015-b989fa6d-e1d1-47fa-8daa-e88a33fd0960.PNG)

Reply:
> I think the filter function is pretty self-documenting and only used once, it's literally just figuring out which vaults should be rendered. So I'm not too eager to make it its own function, but open to change that.

SGTM.

> I'm not sure exactly what you mean by the O(1) table lookup, would you mind spelling out an example?

<img width="738" alt="Screen Shot 2022-04-01 at 7 54 31 AM" src="https://user-images.githubusercontent.com/21505/161289141-8ef854ad-b06e-419b-bda3-70b52b791c14.png">


> We need to preprocess this list to know its length for the rendering logic below, which would be O(n).

I meant the runtime of the function being passed to filter.

> Array.includes is actually pretty fast on two elements:

Definitely. The speed gain from the object isn't worth the typing. The value is more in naming.

```
  const vaultsToRender = vaultsList.filter(([_, {status}]) =>
    showClosed ? renderAlways[status] : renderOpen[status]
```

Again, a "consider" not a requested change.

Reply:
Cleaned this function up a bit as discussed